### DISCUSSION TITLE
Report actual resource usage checked by constraints

### DISCUSSION THREAD

**Discussion by munichmule** [sebastienros/jint#1993] [Q&A]
Hi Guys,

The constraints are great, but I'd like to measure and log actual resource usage for existing scripts, to end up with some reasonable values for constraints. Is there a convenient way to read the values for executed script? I can write some custom "constraints" to collect statistics, but it doesn't sound right - hopefully, you already have some better idea/solution on how to achieve what I need. 

Ideally, I want something like this:
```
var stats = engine.Execute(...).GetStatistics();
logger.Log(stats);
```

Thanks!

**Comment by lahma** [comment]
As most of the constraints rely on shared resources those statistics probably won't make much sense. Only max statements would be reliable from run to run.

**Comment by munichmule** [comment]
@lahma I can measure memory, cpu and total time by myself (hopefully, if we don't switch threads on linux, haha), and statements count with a dummy constraint. 
What I can't measure is recursion depth and, most importantly, max array size. I'm especially interested in the latter one, because, potentially, user can assign very large arrays and eat lots of memory/sequential address space. The memory constrain doesn't help here because it only measures total allocations, but not amount of memory being in use.

**Reply by lahma** [reply]
Maybe you could propose a PR that would update such seen max recursion depth/statements count/array size to new `engine.Advanced.Statistics`?
